src/metadata.cpp

```diff
@@ -61,16 +62,19 @@ bool IMetadata::getStringToRef(const std::string &name,
 	if (!raw)
 		return false;
 
-	const std::string &resolved = resolveString(*raw, &str, recursion);
+	const std::string &resolved = resolveString(*raw, &str, recursion, true);
 	if (&resolved != &str)
 		str = resolved;
 	return true;
 }
 
 const std::string &IMetadata::resolveString(const std::string &str, std::string *place,
-		u16 recursion) const
+		u16 recursion, bool deprecated) const
 {
 	if (recursion <= 1 && str.substr(0, 2) == "${" && str[str.length() - 1] == '}') {
+		if (deprecated)
+			warningstream << "Deprecated use of recursive resolution syntax in metadata: "
+					<< fastWriteJson(Json::Value(str)) << std::endl;
```

Comment:
Why JSON?

Reply:
The intent was to avoid printing arbitrary strings, but I see now that `safe_print_string` exists for this purpose.